OpenChange/MAPIStoreContactsMessage.m

```diff
@@ -1373,4 +1375,25 @@ - (void) save:(TALLOC_CTX *) memCtx
   [self updateVersions];
 }
 
+- (void) save:(TALLOC_CTX *) memCtx
+{
+  NSString *messageClass = [properties objectForKey: MAPIPropertyKey(PR_MESSAGE_CLASS_UNICODE)];
+  if (!messageClass) 
+    {
+      [self logWithFormat: @"ERROR: No message class in ContactMessage. Not saving anything"];
+      return;      
+    }
+
+  if ([messageClass isEqualToString: @"IPM.Contact"])
+    [self saveContact: memCtx]; 
+  else if ([messageClass isEqualToString: @"IPM.DistList"])
+    [self saveDistList: memCtx];
+  else 
+    {
+      [self logWithFormat: @"ERROR: Incorrect message class %@ ContactMessage. Not saving anything", messageClass];
+      return;
```

Comment:
As explained in above comments, I would replace this `else` clause with current `saveContact` method. In this way, we are only modifying the expected behaviour we want to change with this PR, DistList messages are not saved.